src/ImageSharp/Common/Helpers/Numerics.cs

```diff
@@ -532,24 +519,49 @@ public static void Premultiply(Span<Vector4> vectors)
         }
     }
 
+    /// <summary>
+    /// Reverses the result of premultiplying a vector via <see cref="Premultiply(ref Vector4)"/>.
+    /// </summary>
+    /// <param name="source">The <see cref="Vector4"/> to premultiply</param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void UnPremultiply(ref Vector4 source)
+    {
+        Vector4 alpha = PermuteW(source);
+        UnPremultiply(ref source, alpha);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void UnPremultiply(ref Vector4 source, Vector4 alpha)
+    {
+        if (alpha == Vector4.Zero)
+        {
+            return;
+        }
+
+        // Divide source by alpha if alpha is nonzero, otherwise set all components to match the source value
+        // Blend the result with the alpha vector to ensure that the alpha component is unchanged
+        source = WithW(source / alpha, alpha);
```

Comment:
Branching in SIMD code is "generally" undesirable.

It's probably worth comparing this to an implementation that does `CompareEqual` + `Blend` instead (much like you're already doing for the `Avx2` path).

You should be able to use the `AsVector128()` again. With `Sse.CompareEqual` and using `Sse41.BlendVariable`.

If `Sse41` isn't supported (which is unlikely since its been around since 2007), you can replace it with `Sse.Or(Sse.And(left, mask), Sse.AndNot(mask, right))` 

-- That is: `(left & mask) | (right & ~mask)`
-- `Sse.AndNot(x, y)` does `~x & y`, despite its name implying `x & ~y`; a weird quirk of the instruction

Reply:
-- `left` here represents the value to use if the comparison result (`mask`) was true for that element
-- `right` correspondingly is the value to use if it was `false`

Reply:
Would you believe the version I had was actually faster!?

UnPremultiply using Vector4 and branching.

|                Method |     Mean |     Error |    StdDev | Ratio |
|---------------------- |---------:|----------:|----------:|------:|
| UnPremultiplyBaseline | 1.545 us | 0.0070 us | 0.0066 us |  1.00 |
|         UnPremultiply | 1.827 us | 0.0039 us | 0.0037 us |  1.18 |
|     UnPremultiplyBulk | 1.267 us | 0.0033 us | 0.0027 us |  0.82 |

UnPremultiply using Sse41and branching.

|                Method |     Mean |     Error |    StdDev | Ratio |
|---------------------- |---------:|----------:|----------:|------:|
| UnPremultiplyBaseline | 1.546 us | 0.0077 us | 0.0068 us |  1.00 |
|         UnPremultiply | 2.058 us | 0.0043 us | 0.0036 us |  1.33 |
|     UnPremultiplyBulk | 1.268 us | 0.0025 us | 0.0022 us |  0.82 |

Reply:
Yeah. We'd only ever use in for single operations or for overlaps in the AVX run.